### PR TITLE
[T4A4][T09-A3] Tang Di Feng

### PR DESCRIPTION
**Pull request by e0011840**

### CODE REVIEW

File: src/seedu/addressbook/data/person/Person.java [hunk 1]
```diff
@@ -83,5 +83,14 @@ public int hashCode() {
     public String toString() {
         return getAsTextShowAll();
     }
+    
+    public String getPrintableString(Printable...printables) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for(Printable print : printables) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Person.java:89]: may be consider a better name than `print`?

File: src/seedu/addressbook/data/person/Person.java [hunk 2]
```diff
@@ -83,5 +83,14 @@ public int hashCode() {
     public String toString() {
         return getAsTextShowAll();
     }
+    
+    public String getPrintableString(Printable...printables) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Person.java:87]: you can add a header comment to this method as the behavior is different as compared to the one inside the `Printable` interface

File: src/seedu/addressbook/data/person/Printable.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Interface for printing Name, Phone, Email, Address
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Printable.java:4]: this comment is quite off the point, no?
any class that implements this interface can provide the printable string form of the data contained in it.
header comment need to be precise for interfaces and abstract classes, as these form a contract that you are providing to client code developers

File: src/seedu/addressbook/data/person/Printable.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+ */
+
+public interface Printable {
+    
+    public String getPrintableString();
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Printable.java:9]: need comments for this method signature as well

**Review by okkhoy — COMMENTED** [review]
good attempt :+1: 
please take care of the comments in general

**Comment by okkhoy** [comment]
@e0011840 
Some comments added. Please ack & close the PR after reading comments.

**Comment by e0011840** [comment]
Noted With Thanks!

**Pull request closed without merging**